### ISSUE TITLE
{Linked:Bug8885468}[Adaptive cards]: Ensures select element has an accessible name.

### ISSUE THREAD

**Issue by palitsourish** [microsoft/AdaptiveCards#7649]
### Target Platforms

Other

### SDK Version

1.5

### Application Name

Adaptive Cards

### Problem Description

Test Environment:
URL: [Designer | Adaptive Cards](https://adaptivecards.io/designer/) (JSON Code attached)
OS Version: 22H2 (OS Build 25251.1010)
Browser: Microsoft Edge Dev Version 105.0.1300.0 (Official build) dev (64-bit) 
Tool: Accessibility Insights for Web

Repro Steps:
Open the above URL with JSON.
Run AI4W Tool.
Observe the issue.
Actual Result:
Selected element does not have an accessible name.

Expected Result:
Selected element should have an accessible name.

Issue: Ensures select element has an accessible name (select-name - https://dequeuniversity.com/rules/axe/4.4/select-name?application=msftAI)
 
Target application: Designer | Adaptive Cards - https://adaptivecards.io/designer/
 
Element path: #__ac-5bb06f6d-cd19-40c7-be66-45f5c5f980d1
 
Snippet: <select class="ac-input ac-multichoiceInput ac-choiceSetInput-compact" id="__ac-5bb06f6d-cd19-40c7-be66-45f5c5f980d1" style="width: 100%; min-width: 0px;">
 
How to fix:
Fix any of the following:
Form element does not have an implicit (wrapped) <label>
Form element does not have an explicit <label>
aria-label attribute does not exist or is empty
aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
Element has no title attribute
Element's default semantics were not overridden with role="none" or role="presentation"
User Experience:
Screen Reader users would be confused if Selected element does not have an accessible name.

MAS Reference:
[Info and Relationships - Liquid (microsoft.com)](https://liquid.microsoft.com/Web/Object/Read/MS.Accessibility/Requirements/01.03.01)

Have feedback to share on Bugs? Please help fill [Trusted Tester Bug Feedback (office.com)](https://nam06.safelinks.protection.outlook.com/?url=https://forms.office.com/pages/responsepage.aspx?id%3Dv4j5cvGGr0GRqy180BHbR5dUqGQgJ9lItv-jqY8F09lURVdYTENIRkY2UzlCR0NKV1E2VUE5OVpDVy4u&data=04%7c01%7cv-apoorva%40microsoft.com%7c551850210d444be9b28308d9691e3dde%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637656400998658942%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c1000&sdata=l6w0SvVcMSvyGnJ6%2B6uSSgd3braF2dZ7/AdBrqqzIR8%3D&reserved=0)

### Screenshots

<img width="1143" alt="MAS1 3 1-Ensures select element has an accessible name" src="https://user-images.githubusercontent.com/95914144/177564485-ace98230-de5d-4b59-a1c0-eaec997a7d43.png">

### Card JSON

```JSON
{
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "type": "AdaptiveCard",
  "version": "1.0",
  "body": [
    {
      "type": "ColumnSet",
      "columns": [
        {
          "type": "Column",
          "width": 2,
          "items": [
            {
              "type": "TextBlock",
              "text": "##CreateTicketFirstInputHeader##",
              "weight": "Bolder",
              "size": "Default",
              "wrap": true,
              "color": "##HeaderColor##"
            },
            {
              "type": "TextBlock",
              "text": "##CreateTicketTitleLabel##",
              "wrap": true
            },
            {
              "type": "Input.Text",
              "id": "Title",
              "placeholder": "##CreateTicketTitlePlaceholder##",
              "value": "##CreateTicketTitleValue##"
            },
            {
              "type": "TextBlock",
              "text": "##CreateTicketDescriptionLabel##",
              "wrap": true
            },
            {
              "type": "Input.Text",
              "id": "Description",
              "placeholder": "##CreateTicketDescriptionPlaceholder##",
              "value": "##CreateTicketDescriptionValue##",
              "isMultiline": true
            },
            {
              "type": "TextBlock",
              "text": "##CreateTicketCompanyCodeLabel##",
              "isVisible": "##CreateTicketCompanyCodeInputIsVisibleValue##"
            },
            {
              "type": "Input.ChoiceSet",
              "id": "CompanyCode",
              "placeholder": "##CreateTicketCompanyCodePlaceholder##",
              "value": "##CreateTicketCompanyCodeValue##",
              "choices": [ "##CompanyCodeChoices##" ],
              "isVisible": "##CreateTicketCompanyCodeInputIsVisibleValue##"
            },
            {
              "type": "TextBlock",
              "text": "##CreateTicketOrgIdLabel##",
              "isVisible": "##CreateTicketOrgIdInputIsVisibleValue##"
            },
            {
              "type": "Input.ChoiceSet",
              "id": "OrgId",
              "placeholder": "",
              "choices": [ "##OrgIdChoices##" ],
              "isVisible": "##CreateTicketOrgIdInputIsVisibleValue##"
            },

            {
              "type": "TextBlock",
              "text": "##CreateTicketCcLabel##",
              "isVisible": "##CreateTicketCcInputIsVisibleValue##"
            },
            {
              "type": "Input.Text",
              "id": "Cc",
              "placeholder": "Ex: example@microsoft.com,example2@microsoft.com ",
              "value": "##CreateTicketCCValue##",
              "isMultiline": true,
              "isVisible": "##CreateTicketCcInputIsVisibleValue##"
            },
            {
              "type": "Input.ChoiceSet",
              "id": "IsAttachment",
              "choices": [
                {
                  "title": "##CreateTicketAttachmentLabel##",
                  "value": "true"
                }
              ],
              "style": "expanded",
              "isMultiSelect": true,
              "isVisible": "##CreateTicketAttachmentIsVisibleValue##"
            }
          ]
        }
      ]
    }
  ],
  "actions": [
    {
      "type": "Action.Submit",
      "id": "SubmitButtonId",
      "title": "##ButtonProceedLabel##"
    }
  ]
}
```

**Comment by paulcam206** [comment]
This is actually an issue with the card payload, which is missing the [`label` property](https://adaptivecards.io/explorer/Input.ChoiceSet.html) (see [VSO bug](https://microsoft.visualstudio.com/OS/_workitems/edit/40339672) for a bit more detail).

**Comment by anna-dingler** [comment]
Updated JSON:
```JSON
{
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "type": "AdaptiveCard",
  "version": "1.0",
  "body": [
    {
      "type": "ColumnSet",
      "columns": [
        {
          "type": "Column",
          "width": 2,
          "items": [
            {
              "type": "TextBlock",
              "text": "##CreateTicketFirstInputHeader##",
              "weight": "Bolder",
              "size": "Default",
              "wrap": true,
              "color": "##HeaderColor##"
            },
            {
              "type": "Input.Text",
              "id": "Title",
              "placeholder": "##CreateTicketTitlePlaceholder##",
              "value": "##CreateTicketTitleValue##",
              "label": "##CreateTicketTitleLabel##"
            },
            {
              "type": "Input.Text",
              "id": "Description",
              "placeholder": "##CreateTicketDescriptionPlaceholder##",
              "value": "##CreateTicketDescriptionValue##",
              "isMultiline": true,
              "label": "##CreateTicketDescriptionLabel##"
            },
            {
              "type": "Input.ChoiceSet",
              "id": "CompanyCode",
              "placeholder": "##CreateTicketCompanyCodePlaceholder##",
              "value": "##CreateTicketCompanyCodeValue##",
              "choices": [
                "##CompanyCodeChoices##"
              ],
              "isVisible": "##CreateTicketCompanyCodeInputIsVisibleValue##",
              "label": "##CreateTicketCompanyCodeLabel##"
            },
            {
              "type": "Input.ChoiceSet",
              "id": "OrgId",
              "placeholder": "",
              "choices": [
                "##OrgIdChoices##"
              ],
              "isVisible": "##CreateTicketOrgIdInputIsVisibleValue##",
              "label": "##CreateTicketOrgIdLabel##"
            },
            {
              "type": "Input.Text",
              "id": "Cc",
              "placeholder": "Ex: example@microsoft.com,example2@microsoft.com ",
              "value": "##CreateTicketCCValue##",
              "isMultiline": true,
              "isVisible": "##CreateTicketCcInputIsVisibleValue##",
              "label": "##CreateTicketCcLabel##"
            },
            {
              "type": "Input.ChoiceSet",
              "id": "IsAttachment",
              "choices": [
                {
                  "title": "##CreateTicketAttachmentLabel##",
                  "value": "true"
                }
              ],
              "style": "expanded",
              "isMultiSelect": true,
              "isVisible": "##CreateTicketAttachmentIsVisibleValue##"
            }
          ]
        }
      ]
    }
  ],
  "actions": [
    {
      "type": "Action.Submit",
      "id": "SubmitButtonId",
      "title": "##ButtonProceedLabel##"
    }
  ]
}
```

**Comment by vagpt** [comment]
This issue is not repro at our end, hence adding A11yTTValidated tag.
